### ISSUE TITLE
Default `-j 2` value hurting performance

### ISSUE THREAD

**Issue by GitHubGeek** [jessek/hashdeep#325]
TL;DR: **-j** should have a sane default of 1 thread for checksumming and matching operations

Running md5deep 3.9.2 on Ubuntu 14.04. I noticed the during matching mode (`md5deep -x md5.txt -r /some/dir`) the I/O thoughtput is way below what my disk is capable of (25MB/s vs 160MB/s). In `iotop` I can see that 2 `md5deep` threads are utilising the bandwidth.

Adding `-j1` to the command solves the problem.

I can't see the benefits of multi-threading when checksumming / matching files on a single disk. Modern CPUs can crunch MD5 checksums in tens of GB/s. So, unless the user has an exotic multi-SSD RAID-0 setup, the bottleneck is in the disks and multi-threading would not improve performance. Moreover, performance on mechanical HDDs would almost certainly suffer when the threads are competing for reads and causing excessive seeks.

Probably related to #298

EDIT: Done some more testing, this issue applies to checksumming as well.

**Comment by dima-stefantsov** [comment]
Today, in 2023, I installed apt install md5deep (hashdeep) in Ubuntu 22.04 server, and behaviour is still the same: on a 24thread server md5deep -r /dir/ gives 30 MB/s, and same command with -j0 (same for -j1) gives ~200 MB/s.
A weird default.
